components/life-cycle-extension/src/main/java/com/wso2telco/dep/lifecycleextension/notification/impl/NotificationServiceImpl.java

```diff
@@ -87,4 +86,48 @@ public void sendApiProviderEmail(RequestContext context) {
             log.error("Failed to validate user details for send email ", e);
         }
     }
-}
+
+    @Override
+    public boolean validateIsNotNewApiVersion(RequestContext context) {
+
+        try {
+
+            API api = getApiDetails(context);
+
+            String provider = APIUtil.replaceEmailDomain(api.getId().getProviderName());
+            APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(provider);
+            List<API> oldApiList = apiProvider.getAPIsByProvider(provider);
+
+            if(!oldApiList.isEmpty()){
+                APIVersionComparator versionComparator = new APIVersionComparator();
+                for (API oldApi : oldApiList) {
+                    if(oldApi.getId().getApiName().equalsIgnoreCase(api.getId().getApiName()) && versionComparator.compare(oldApi, api) < 0 &&
+                            (APIConstants.PUBLISHED.equals(oldApi.getStatus()))){
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }catch (Exception e){
+            log.error("Failed to validate api version details ", e);
+            return false;
+        }
+    }
+
+    private API getApiDetails(RequestContext context)throws Exception{
+
+        try {
+            GenericArtifactManager artifactManager = APIUtil.getArtifactManager(context.getSystemRegistry(),
+                    APIConstants.API_KEY);
+            String artifactId = context.getResource().getUUID();
+            GenericArtifact apiArtifact = artifactManager.getGenericArtifact(artifactId);
+            return APIUtil.getAPI(apiArtifact);
+
+        }catch (Exception e){
+            log.error("Failed to get request context api details ", e);
+            throw e;
```

Comment:
No need both log and rethrow again. Remove the log

Reply:
@sachindakumara you can remove this try-catch completely.
https://stackoverflow.com/a/6640029/3728639